generator.ts

```diff
@@ -0,0 +1,254 @@
+import { promises as fsPromises } from 'fs';
+
+import _ from 'lodash';
+import yaml from 'js-yaml';
+
+import { init } from './init';
+import {
+  getResourceSchema,
+  getResultSchema,
+  getSetResultSchema,
+  getPostBodySchema,
+  getPatchBodySchema,
+} from './schemas';
+
+/**
+ * Gets the prefix of the resource schema in components/schemas
+ *
+ * @param resourceName
+ * @returns The prefix
+ */
+const getResourceSchemaPrefix = (resourceName: string) => _.capitalize(resourceName);
+
+/**
+ * Gets the method for a given endpoint. Example: `getById` -> `get`
+ *
+ * @param endpoint
+ * @returns The method
+ */
+const getEndpointMethod = (endpoint: string): string => endpoint.match(/^[a-z]+/)![0];
+
+/**
+ * Generates the value for `operationId`
+ *
+ * @param endpoint
+ * @param resourceName
+ * @param resource - The resource object
+ * @returns The value of operationId
+ */
+const getOperationId = (endpoint: string, resourceName: string, resource: any): string => {
+  if (endpoint === 'get') {
+    return `${endpoint}${_.capitalize(resource.plural)}`;
+  }
+  const words = _.words(endpoint, /(^[a-z]+)|([A-Z][a-z]*)/g);
+  words.splice(1, 0, `${_.capitalize(resourceName)}`);
+  return _.join(words, '');
+};
```

Comment:
Wow, I'm very impressed on how you accomplished this 👍 
I guess this function can be simplified a bit by using this package: https://www.npmjs.com/package/camelcase

Reply:
Hm, wouldn't I still need to call `_.words`, `splice`, and `_.join` in order to insert the resource name into the operationId? I could see replacing `_.capitalize` with `camelCase` but I think this function can already assume that `resourceName` is in camelCase. If not, I should probably add this check beforehand somewhere. Also, it looks like lodash has a `_.camelCase` function as well https://lodash.com/docs/#camelCase.

Reply:
@ianbrown9475 yeah, that's what I meant. I was mainly saying we can just call camelCase package once to replace `_.captialize` (I didn't know lodash also has a method for transforming string to camelCase, good find).